### PR TITLE
[ospp] add push style collector

### PR DESCRIPTION
**Pull request by vinci-897**
## What's changed?

add push style collector.

## Checklist

- [x]  I have read the [Contributing Guide](https://hertzbeat.com/docs/others/contributing/)
- [x]  I have written the necessary doc or comment.
- [x]  I have added the necessary unit tests and all cases have passed.

### CODE REVIEW

File: manager/src/main/resources/application.yml [hunk 1]
```diff
@@ -70,16 +70,17 @@ spring:
...
 
   datasource:
-    driver-class-name: org.h2.Driver
-    username: sa
-    password: 123456
-    url: jdbc:h2:./data/hertzbeat;MODE=MYSQL
+    driver-class-name: com.mysql.cj.jdbc.Driver
+    username: root
+    password: 12345678
+    url: jdbc:mysql://127.0.0.1:3306/hbtest?useUnicode=true&characterEncoding=UTF-8&serverTimezone=Asia/Shanghai&useSSL=false
```
Comment (Ceilzcx) [inline manager/src/main/resources/application.yml:76]: datasource这部分代码不要push上来

File: common/src/main/java/org/dromara/hertzbeat/common/entity/push/PushMetrics.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import javax.persistence.*;
+
+/**
+ * push metrics entity
+ *
+ * @author vinci
+ */
+@Entity
+@Table(name = "hzb_pushmetrics", indexes = {
```
Comment (Ceilzcx) [inline common/src/main/java/org/dromara/hertzbeat/common/entity/push/PushMetrics.java:17]: hzb_pushmetrics改成hzb_push_metrics

File: common/src/main/java/org/dromara/hertzbeat/common/entity/push/PushMetricsDto.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @Data
+    @Builder
+    @AllArgsConstructor
+    @NoArgsConstructor
+    public static class Metrics {
+        private long monitorId;
+        private Long time;
+        private List<String> metrics;
```
Comment (Ceilzcx) [inline common/src/main/java/org/dromara/hertzbeat/common/entity/push/PushMetricsDto.java:34]: push的数据格式不应该是Map的吗？key表示字段名，value表示值

**Review by Ceilzcx — APPROVED** [review]

**Comment by hertzbeat** [comment]
> Bot detected the issue body's language is not English, translate it automatically. 👯👭🏻🧑‍🤝‍🧑👫🧑🏿‍🤝‍🧑🏻👩🏾‍🤝‍👨🏿👬🏿         
----    
image

**Comment by tomsun28** [comment]
👍

File: manager/src/main/resources/define/app-push.yml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    # required-true or false
+    # required-是否是必输项 true-必填 false-可选
+    required: true
+    # field-param field key
+# collect metrics config list
+# 采集指标组配置列表
+metrics:
+  # metrics - all
+  # 监控指标组 - all
+  - name: all
```
Comment (Ceilzcx) [inline manager/src/main/resources/define/app-push.yml:99]: 那么改下吧，就叫metrics好了

File: push/src/main/java/org/dromara/hertzbeat/push/service/impl/PushServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    @Autowired
+    private PushMonitorDao monitorDao;
+
+    @Autowired
+    private PushMetricsDao metricsDao;
+
+    @Override
+    public void pushMetricsData(PushMetricsDto pushMetricsDto) throws RuntimeException {
+        List<PushMetrics> pushMetricsList = new ArrayList<>();
+        // TODO: 改成只取不同的monitorId
```
Comment (Ceilzcx) [inline push/src/main/java/org/dromara/hertzbeat/push/service/impl/PushServiceImpl.java:37]: 这里是要改，因为你monitorId是放在里面的，加一个缓存也好，放一个Map里面，不然查表次数太多了

File: collector/src/main/java/org/dromara/hertzbeat/collector/collect/push/PushCollectImpl.java [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+//        });
+//        if (jsonMap == null) {
+//            throw new NullPointerException("parse result is null");
+//        }
+        PushMetricsDto pushMetricsDto = JsonUtil.fromJson(resp, new TypeReference<Message<PushMetricsDto>>() {
+        }).getData();
+        if (pushMetricsDto == null) {
+            throw new NullPointerException("parse result is null");
+        }
+        //TODO: 由于collectRep里只有一个时间字段，没办法保存每一条记录被push的时间
```
Comment (Ceilzcx) [inline collector/src/main/java/org/dromara/hertzbeat/collector/collect/push/PushCollectImpl.java:148]: 因为把metricsList拼成一个返回了，导致time字段用不上了，这里时间就有点问题了

**Comment by hertzbeat** [comment]
> Bot detected the issue body's language is not English, translate it automatically. 👯👭🏻🧑‍🤝‍🧑👫🧑🏿‍🤝‍🧑🏻👩🏾‍🤝‍👨🏿👬🏿         
----    
The code logic has been roughly modified according to the requirements mentioned by Tom last week, the database has been cached, and some fragmentary todos have been completed.
At present, the database still uses jpa, which is not connected to the warehouse and cannot use the time series database. If there is no need to rush to complete this recently, I can continue to do it after ospp.

**Comment by hertzbeat** [comment]
> Bot detected the issue body's language is not English, translate it automatically. 👯👭🏻🧑‍🤝‍🧑👫🧑🏿‍🤝‍🧑🏻👩🏾‍🤝‍👨🏿👬🏿         
----    
Brother Zheng and Brother Tom said during the discussion last week that due to the time field design and abstractCollect interface design of collectRep, it is currently impossible to collect multiple records at the same time, so a compromise solution was used. If I consider modifying this later, I can continue to finish it. After all, I am quite familiar with this part of the code.

**Review by Ceilzcx — APPROVED** [review]

**Review by tomsun28 — APPROVED** [review]
LGTM👍👍

**Pull request merged**